### DISCUSSION TITLE
Chromiun don't auto start on raspi 4 and bookworm

### DISCUSSION THREAD

**Discussion by laguinch** [photobooth-app/photobooth-app#116] [Q&A] [answered]
Hy, the auto lunch from Photo Booth-app in kiosk mode don't start. A chromium page is open but not in kiosk mode. In seems that the tutorial is different for bookworm see on https://forums.raspberrypi.com/viewtopic.php?t=358317

**Comment by mgineer85** [comment]
could you make it work? Seems there are some ways to get it work.

**Reply by laguinch** [reply]
Ok i would try. I just need to remove the 2 files i have create like your tutorial ? And try to modify the wayfire file that’s it ?

**Reply by mgineer85** [reply]
I install a bookworm system again to create updated installation procedure.   I would not recommend to go back to X11.

**Reply by laguinch** [reply]
I will wait your update installation it’s more safe for me 😉. Thx

**Reply by mgineer85** [reply]
Pls check the updated instructions: https://mgrl.github.io/photobooth-docs/setup/installation/#bookworm-platform

It seems you are getting closer to a full system? If you like to I would be glad if you create a new discussion describing your setup and post some photos.   We could add it here also if you like to:
https://mgrl.github.io/photobooth-docs/projects/

**Reply by laguinch** [reply]
i will do the update tomorrow evening after my job. And I will create a new discussion with my all configuration, 3D print,.......
A question any more, it's better I remove the ~/Desktop/photobooth-app.desktop file I create before ?

**Reply by mgineer85** [reply]
Delete the old one and create a new for bookworm.

I recommend to create the desktop icon also - just in case you have it available:
- if browser crashes, you are able to restart even on an event.
- you can close the browser any time by using the keyboard combination ALT + F4; with the icon it's easy to restart.

**Reply by laguinch** [reply]
Hy, i test the modification.
To access the wayfire file the command is :  .config/wayfire.ini and not  ~/config/wayfire.ini
I make the changes but with chromium it's not running with with firefox it's ok. I don't understand why ?

**Reply by laguinch** [reply]
in the installation command from the ~/Desktop/photobooth-app.desktop there is an error in the line Exec=chromium-browser--kiosk  there is space to adjust after browser Exec=chromium-browser --kiosk

**Reply by laguinch** [reply, marked as answer]
to run with chromium I test to add chromium = chromium-browser--kiosk --noerrdialogs --disable-infobars --no-first-run --ozone-platform=wayland --enable-features=OverlayScrollbar --start-maximized http://localhost:8000/ and it's work

**Reply by mgineer85** [reply]
Mistakes fixed in next docs update.

**Reply by laguinch** [reply]
hy, you don't fix the error space in this line command after "browser" in the ~/Desktop/photobooth-app.desktop file -> Exec=chromium-browser--kiosk there is space to adjust after browser Exec=chromium-browser --kiosk

**Reply by mgineer85** [reply]
I don't think that chromium-browser--kiosk is correct. --kiosk is an argument.

**Reply by laguinch** [reply]
yes, chromium-browser--kiosk isn't correct a space must be add between browser and --Kiosk

**Reply by f2103** [reply]
The tutorial as written in the setup page didn't work for my either on rpi3b using bookworm.

What has worked for me:
`sudo apt-get install --no-install-recommends xserver-xorg x11-xserver-utils xinit openbox`
`sudo apt-get install --no-install-recommends chromium-browser`
`sudo nano /etc/xdg/openbox/autostart`

```
# Disable any form of screen saver / screen blanking / power management
xset s off
xset s noblank
xset -dpms

# Allow quitting the X server with CTRL-ATL-Backspace
setxkbmap -option terminate:ctrl_alt_bksp

# Start Chromium in kiosk mode
sed -i 's/"exited_cleanly":false/"exited_cleanly":true/' ~/.config/chromium/'Local State'
sed -i 's/"exited_cleanly":false/"exited_cleanly":true/; s/"exit_type":"[^"]\+"/"exit_type":"Normal"/' ~/.config/chromium/Default/Preferences
chromium-browser --check-for-update-interval=1 --simulate-critical-update --disable-infobars --kiosk 'http://localhost:8000'
```

Edit:
`sudo nano /etc/X11/Xwrapper.config`

The file should be empty so place in the first line just this:
`allowed_users = anybody`

Exit and save with STRG + X, confirm the saving with Y.
Also set the rights of Xorg differently and add user pi to the tty group:

`sudo chmod u+s /usr/bin/Xorg`
`sudo usermod --append --groups tty,input $(whoami)`

`sudo update-alternatives --config x-session-manager` -> select openbox

`sudo nano /home/$(whoami)/.bash_profile`
and add following text:
`sudo DISPLAY=:0 startx -- -nocursor`

Yet this is still not perfect because chrome starts when the webserver is not available, but it refreshed automatically later. Let me know if you guys have any idea for a delay or additional splash screen.

[based on tutorial from https://florianmuller.com/raspberry-pi-photobooth-in-a-classic-vintage-plate-camera -> many thanks!]

**Reply by mgineer85** [reply]
What was the initial issue that led you to install x11 again instead using wayland?
